src/com/amazon/corretto/crypto/provider/XDHGen.java

```diff
@@ -0,0 +1,68 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGeneratorSpi;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+class XDHGen extends KeyPairGeneratorSpi {
+  // Generates a new X25519 key and returns a pointer to it encoded as a java |long|
+  private static native long generateEvpXECKey();
+
+  private final AmazonCorrettoCryptoProvider provider_;
+  private KeyFactory kf;
+
+  XDHGen(AmazonCorrettoCryptoProvider provider) {
+    Loader.checkNativeLibraryAvailability();
+    provider_ = provider;
+    try {
+      /* Though XEC keys were introduced in JDK11, while checking key size against the spec parameters,
+       * JDK 11 XDHPrivateKeyImpl checks the wrapped key which is 34 bytes long
```

Comment:
```suggestion
       * JDK 11 XDHPrivateKeyImpl checks the encoded key which is 34 bytes long
```

small nit -- "wrapped" in the context of keys generally refers to [encrypting key material](https://csrc.nist.gov/pubs/sp/800/38/f/final).